hack/generator/pkg/jsonast/testdata/OneOf/OneOf_generates_wrapper_type_with_marshal_helper.golden

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// This is a generated file. Please do not make manual changes.
+package v20200101
+
+import (
+	"encoding/json"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+/*Generated from: https://test.test/schemas/2020-01-01/test.json#/definitions/Bar*/
+type Bar struct {
+	// +kubebuilder:validation:Required
+	Size int `json:"size"`
+}
+
+/*Generated from: https://test.test/schemas/2020-01-01/test.json#/definitions/Baz*/
+type Baz struct {
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+}
+
+/*Generated from: https://test.test/schemas/2020-01-01/test.json#/definitions/Foo*/
+type Foo struct {
+	Name *string `json:"name"`
+}
+
+/*Generated from: https://test.test/schemas/2020-01-01/test.json*/
+type Test struct {
+	Bar *Bar `json:"bar"`
+	Baz *Baz `json:"baz"`
+	Foo *Foo `json:"foo"`
+}
+
+// MarshalJSON marshals the object as JSON
```

Comment:
Is there anything more meaningful we can include in this doc comment?

Reply:
Possibly - I realize this is a bit of a cop-out comment but unsure what else to say really.
We could make the documentation specific to the implementation by saying: `MarshalJSON defers marshaling to the first non-nil property to simulate OneOf`?

Do you have any suggestions?

Reply:
`MarshalJSON defers JSON marshaling to the first non-nil property, because TestProperties represents a discriminated union (JSON OneOf)`

It's an improvement I think.